framework/src/media/audio/audio_manager.c

```diff
@@ -123,15 +129,37 @@ audio_manager_result_t find_input_audio_card(void)
 			|| (dir_entry->d_name[1] != 'c')
 			|| (dir_entry->d_name[2] != 'm')
 			|| (sscanf(&dir_entry->d_name[3], "C%uD%u%c", &card_id, &device_id, &type) != 3)) {
-			closedir(dir_info);
-			return AUDIO_MANAGER_INVALID_DEVICE_NAME;
+			continue;
 		}
 
 		if ((type == 'c') && (card_id < CONFIG_AUDIO_MAX_INPUT_CARD_NUM) && (device_id < MAX_AUDIO_DEVICE_NUM)) {
+#ifndef CONFIG_AUDIO_EXCLUDE_VOLUME
+			struct audio_caps_desc_s caps_desc;
+			FAR struct audio_caps_s *caps = &(caps_desc.caps);
+			int fd = open(strcat(path, dir_entry->d_name), O_RDONLY);
```

Comment:
This logic has a problem. What if there are 2 or more devices, 
The 'path' variable will be accumulated always.
Isn't it?

Reply:
Right. "path" is now declared inside while loop to avoid the accmulation.